rocketpy/Function.py

```diff
@@ -2118,77 +2195,78 @@ def __sub__(self, other):
                 and isinstance(self.source, np.ndarray)
                 and self.__interpolation__ == other.__interpolation__
```

Comment:
The git diff is a bit weird due to staging, so don't base the changes too much from it. Are these checks really needed in order to perform the operations?

- The `self.__inputs__` I believe is made to ensure unit consistency. However, that only seems reasonable if there were a `unit` parameter for the `Function`, since the `__inputs__` might change or have slightly different names while being the same;

- The interpolation check makes some sense to infer the interpolation of the result, but that seems quite restrictive, wouldn't it be more interesting to not check this and keep the `self` interpolation?